app/src/main/java/com/team7/cmput301/android/theirisproject/activity/AddPatientDialogFragment.java

```diff
@@ -65,7 +66,10 @@ public void onComplete(Boolean success) {
                                 listener.onFinishAddPatient(success);
                             }
                         };
-                        controller.addPatient(addPatientEditText.getText().toString(), callback);
+                        Boolean success = controller.addPatient(addPatientEditText.getText().toString(), callback);
```

Comment:
Instead of making every controller method that calls a task return a boolean, would it be better to just check whether or not we have internet in the activities themselves? And if we don't have internet, don't call the controller method?

Reply:
Controllers also require the internet status to decide whether to render local data, put updates in update queue, etc, so it's better to keep the check in the controllers.